analyzers/conversion_overflow.go

```diff
@@ -162,66 +242,263 @@ func isStringToIntConversion(instr *ssa.Convert, dstType string) bool {
 	}
 }
 
-type integer struct {
-	signed bool
-	size   int
+func hasExplicitRangeCheck(instr *ssa.Convert, dstType string) bool {
+	dstInt, err := parseIntType(dstType)
+	if err != nil {
+		return false
+	}
+
+	srcInt, err := parseIntType(instr.X.Type().String())
+	if err != nil {
+		return false
+	}
+
+	minValue := srcInt.min
+	maxValue := srcInt.max
+	explicitPositiveVals := []uint{}
+	explicitNegativeVals := []int{}
+
+	if minValue > dstInt.min && maxValue < dstInt.max {
+		return true
+	}
+
+	visitedIfs := make(map[*ssa.If]bool)
+	for _, block := range instr.Parent().Blocks {
+		for _, blockInstr := range block.Instrs {
+			switch v := blockInstr.(type) {
+			case *ssa.If:
+				result := getResultRange(v, instr, visitedIfs)
+				if result.isRangeCheck {
+					minValue = max(minValue, &result.minValue)
+					maxValue = min(maxValue, &result.maxValue)
+					explicitPositiveVals = append(explicitPositiveVals, result.explixitPositiveVals...)
+					explicitNegativeVals = append(explicitNegativeVals, result.explicitNegativeVals...)
+				}
+			case *ssa.Call:
+				// These function return an int of a guaranteed size.
+				if v != instr.X {
+					continue
+				}
+				if fn, isBuiltin := v.Call.Value.(*ssa.Builtin); isBuiltin {
+					switch fn.Name() {
+					case "len", "cap":
+						minValue = 0
+					}
+				}
+			case *ssa.Convert:
+				if v == instr {
+					break
+				}
```

Comment:
This case does nothing and can be omitted. The `break` statement exits the switch, which will be exited regardless.